econml/dml.py

```diff
@@ -42,68 +71,130 @@ class _RLearner(LinearCateEstimator):
         take an extra second argument (the treatment residuals).  Predict, on the other hand,
         should just take the features and return the constant marginal effect.
 
-    n_splits: int
-        The number of splits to use when fitting the first-stage models.
+    n_splits: int, or a custom cross validator object
```

Comment:
Can we maybe use the exact same interface that sklearn uses for all their cross validated classes, e.g.:
```README
cv : int, cross-validation generator or an iterable, optional
  - Determines the cross-validation splitting strategy. Possible inputs for cv are:
  - None, to use the default 3-fold cross-validation,
  - integer, to specify the number of folds.
  - CV splitter,
  - An iterable yielding (train, test) splits as arrays of indices.
```
see e.g. here [LassoCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LassoCV.html)

All these interfaces seem useful. For instance in the weighted application, most prob the last interface is the right one.

Reply:
We could for instance directly use the "check_cv" utility function from scikit learn which implements exactly this logic:
https://github.com/scikit-learn/scikit-learn/blob/1495f69242646d239d89a5713982946b8ffcf9d9/sklearn/model_selection/_split.py#L1940

Reply:
This is now handled via `check_cv`.